packages/caliper-fabric/lib/connector-versions/peer-gateway/PeerGateway.js

```diff
@@ -0,0 +1,442 @@
+
+        //create signer using the private key of the peer
+        const privateKey = crypto.createPrivateKey(walletIdentity.privateKey);
+        const signer = signers.newPrivateKeySigner(privateKey);
+
+        //create gateway instance using the grpc client, identity and signer
+        const gateway = connect({
+            client,
+            identity,
+            signer,
+            evaluateOptions: PeerGateway._configDefaultTimeout,
+            endorseOptions: PeerGateway._configDefaultTimeout,
+            submitOptions: PeerGateway._configDefaultTimeout,
+            commitStatusOptions: PeerGateway._configDefaultTimeout
+        });
+
+        logger.debug('Exiting _createGatewayWithIdentity');
+
+        return gateway;
+    }
+
+    /**
+     * Submit or evaluate a transaction
+     * @param {FabricRequestSettings} invokeSettings The settings associated with the transaction submission.
+     * @param {boolean} isSubmit boolean flag to indicate if the transaction is a submit or evaluate
+     * @return {Promise<TxStatus>} The result and stats of the transaction invocation.
+     * @async
+     */
+    async _submitOrEvaluateTransaction(invokeSettings, isSubmit) {
+        const smartContract = await this._getContractForIdentityOnChannelWithChaincodeID(invokeSettings.invokerMspId, invokeSettings.invokerIdentity, invokeSettings.channel, invokeSettings.contractId);
+
+        // Build the Caliper TxStatus, this is a reduced item when compared to the low level API capabilities
+        // - TxID is not available until after transaction submit/evaluate and must be set at that point
+        const invokeStatus = new TxStatus();
+
+        // set the proposal Options(arguments, endorsingOrganizations, transientData)
+        const proposalOptions = {};
+        // add contract arguments to proposal Options
+        proposalOptions.arguments = invokeSettings.contractArguments;
+        // Add transient data if present
+        if (invokeSettings.transientMap) {
+            const transientData = {};
+            const keys = Array.from(Object.keys(invokeSettings.transientMap));
+            keys.forEach((key) => {
+                transientData[key] = Buffer.from(invokeSettings.transientMap[key]);
+            });
+            proposalOptions.transientData = transientData;
+        }
+        // set transaction invocation result to return
+        try {
+            const proposal = smartContract.newProposal(invokeSettings.contractFunction, proposalOptions);
+            invokeStatus.Set('time_create', Date.now());
+            invokeStatus.SetID(proposal.getTransactionId());
+            if (isSubmit) {
+                invokeStatus.Set('request_type', 'transaction');
+                const transaction = await proposal.endorse();
+                const subtx = await transaction.submit();
+                const status = await subtx.getStatus();
+                if (status.successful){
```

Comment:
For readability it would be better to do
```javascript
if (!status.successful) {
   throw Error(`Failed to submit trasaction with status code: ${status.code}`);
}

invokeStatus.SetStatusSuccess();
invokeStatus.SetResult(subtx.getResult());
```